src/data/set/intervals/ord_connected.lean

```diff
@@ -94,6 +94,21 @@ lemma ord_connected_empty : ord_connected (∅ : set α) := λ x, false.elim
 
 lemma ord_connected_univ : ord_connected (univ : set α) := λ _ _ _ _, subset_univ _
 
+/-- In a dense order `α`, the subtype from an `ord_connected` set is also densely ordered. -/
+lemma subtype_densely_ordered [densely_ordered α] {s : set α} (hs : ord_connected s) :
+  densely_ordered s :=
+⟨ begin
+    intros a₁ a₂ ha,
+    have ha' : ↑a₁ < ↑a₂ := ha,
+    obtain ⟨x, ha₁x, hxa₂⟩ := dense ha',
+    refine ⟨⟨x, _⟩, ⟨ha₁x, hxa₂⟩⟩,
+    exact (hs a₁.2 a₂.2) (Ioo_subset_Icc_self ⟨ha₁x, hxa₂⟩),
+  end ⟩
+
+/-- In a dense order `α`, open intervals `Ioo a b` are also densely ordered. -/
+instance [densely_ordered α] {a b : α} : densely_ordered (Ioo a b) :=
```

Comment:
Wouldn't it make sense to have a typeclass for `ord_connected` subsets (instead of a definition), instances that all kinds of intervals are `ord_connected`, and then give `subtype_densely_ordered` as an instance, that would cover at once all intervals?

Reply:
To make the instances that all kinds of intervals are `ord_connected`, I put `attribute [instance]` after the existing lemmas that all kinds of intervals are `ord_connected`.  Let me know if that's not the correct way.

Reply:
Yes, it's fine (or you could replace these lemmas directly with instances, as you prefer).